tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -389,3 +392,152 @@ def finalize(self, state):
+        By default, the dtype is `tf.float32`. Any integer dtypes will be
+        cast to corresponding floats (i.e. `tf.int32` will be cast to
+        `tf.float32`), as intermediate calculations should be performing
+        floating-point division.
+    """
+    self.shape = shape
+    self.moment = moment
+    if dtype is tf.int64:
+      dtype = tf.float64
+    elif dtype.is_integer:
+      dtype = tf.float32
+    self.dtype = dtype
+
+  def initialize(self):
+    """Initializes an empty `RunningCentralMomentsState`.
+
+    Returns:
+      state: `RunningCentralMomentsState` representing a stream of no
+        inputs.
+    """
+    mean_stream = RunningMean(
+        self.shape, self.dtype
+    )
+    return RunningCentralMomentsState(
+        num_samples=0,
+        mean=mean_stream.initialize().mean,
+        sum_exponentiated_residuals=tf.zeros(
+            (self.moment,) + self.shape, self.dtype),
+    )
+
+  def update(self, state, new_sample):
+    """Update the `RunningCentralMomentsState` with a new sample.
+
+    Args:
+      state: `RunningCentralMomentsState` that represents the current
+        state of running statistics.
+      new_sample: Incoming `Tensor` sample with shape and dtype compatible with
+        those used to form the `RunningCentralMomentsState`.
+
+    Returns:
+      state: `RunningCentralMomentsState` with updated calculations.
+    """
+    n_2 = 1
+    n_1 = state.num_samples
+    n = tf.cast(n_1 + n_2, dtype=self.dtype)
+    delta_mean = new_sample - state.mean
+    mean_stream = RunningMean(
+        self.shape, self.dtype
+    )
+    new_mean = mean_stream.update(RunningMeanState(
+        tf.cast(state.num_samples, self.dtype), state.mean,
+    ), new_sample).mean
+    old_res = state.sum_exponentiated_residuals
+    new_sum_exponentiated_residuals = [tf.zeros(self.shape, self.dtype)]
+
+    # the following two nested for loops calculate equation 2.9 in Pebay's
+    # 2008 paper from smallest moment to highest.
+    for p in range(2, self.moment + 1):
+      summation = tf.zeros(self.shape, self.dtype)
+      for k in range(1, p - 1):
+        adjusted_old_res = ((-delta_mean / n) ** k) * old_res[p-k-1]
+        summation += self._n_choose_k(p, k) * adjusted_old_res
+      # the `adj_term` refers to the final term in equation 2.9 and is not
+      # transcribed exactly, but rather, it's actually simplified to avoid
+      # having a `(n-1)` denominator.
+      adj_term = (((delta_mean / n) ** p) * (n - 1) *
+                  ((n - 1) ** (p - 1) + (-1) ** p))
+      new_sum_pth_residual = old_res[p - 1] + summation + adj_term
+      new_sum_exponentiated_residuals.append(new_sum_pth_residual)
+
+    return RunningCentralMomentsState(
+        num_samples=n_1 + 1,
+        mean=new_mean,
+        sum_exponentiated_residuals=tf.convert_to_tensor(
+            new_sum_exponentiated_residuals
+        )
+    )
+
+  def finalize(self, state, return_all_moments=True):
```

Comment:
I don't know if this arg really pulls its weight. The user can just as easily write the `r.finalize(state)[r.moment - 1]` themselves.

Reply:
Yeah, it's not a critical argument to have -- I just thought it would be a nice convenience. Do you think it's better to remove it for API simplicity?